packages/protocol-http/src/httpRequest.ts

```diff
@@ -44,21 +165,31 @@ export class HttpRequest implements HttpMessage, Endpoint {
   }
 
   clone(): HttpRequest {
-    const cloned = new HttpRequest({
+    return HttpRequest.from({
       ...this,
-      headers: { ...this.headers },
+      destination: new URL(this.destination),
+      fields: Fields.from(
+        this.fields.getAll().map((field) => ({
+          ...field,
+          values: [...field.values],
```

Comment:
Perhaps fields should implement clone. Hard to remember to do this every time.

Reply:
Yea I thought about it. I figured we could implement it later if we needed it rather than exposing it now, but I don't have a strong opinion either way.